packages/normalizr/src/denormalize.ts

```diff
@@ -51,25 +47,33 @@ Make sure you do not have multiple versions of @rest-hooks/normalizr installed.`
     return [entity, false, false];
   }
 
+  const pk =
+    typeof id === 'string'
+      ? id
+      : schema.pk(isImmutable(entity) ? (entity as any).toJS() : entity);
+  if (pk === undefined || pk === '') {
+    return [entity, false, false];
+  }
+
   if (localCache[schema.key] === undefined) {
     localCache[schema.key] = {};
   }
 
   let found = true;
   let deleted = false;
 
-  if (!localCache[schema.key][id]) {
+  if (!localCache[schema.key][pk]) {
     const trackingIndex = dependencies.length;
     dependencies.push(entity);
 
     const wrappedUnvisit = withTrackedEntities(unvisit);
     // { [DRAFT] } means we are still processing - which if found indicates a cycle
     wrappedUnvisit.setLocal = entityCopy =>
-      (localCache[schema.key][id] = { [DRAFT]: entityCopy, i: trackingIndex });
+      (localCache[schema.key][pk] = { [DRAFT]: entityCopy, i: trackingIndex });
```

Comment:
@ljharb since 'pk' here can be derived from networking response, is prototype pollution a potential issue here? The codeql scan just complains about schema.key - but since that's developer set it does not concern me.

Reply:
Yes, unless `localCache` is has null objects as its values.

Reply:
Actually if we're dictating the value set.... the best that they could do is break the prototype since they can't control what it's set to - no?